docs/adrs/003-Retry_endpoint_design.md

```diff
@@ -32,58 +31,75 @@ The endpoint takes a job or list of job ids and then attempts to resubmit them t
+* Adds `child_job_id` to `parent_job_id.child_job_ids[]`
 
 ## Retry_job behavior
-* Blocking and single submit to HTCondor. It should be fine
+* Blocking and single submit to HTCondor. It should be fine as it returns relatively quickly
   
 ## Retry_jobs behavior
-* Submitting multiple jobs uses the `run_job` endpoint, and is blocking (NOT OK!)
+* Submitting multiple jobs for retry serially calls the same code path
+used for running a single job and blocks until all jobs have been
+submitted to the condor queue. This can cause issues if the
+network drops, and makes the narrative not aware of the state of
+the retry. Submitting 100 jobs currently takes 9 seconds, and that
+is a lot of time for things to go wrong. 
+* (Follow up: Hopefully the making the narrative aware of the state of the retry will be mitigated by the narrative backend. It just blocks on the call anyway, with the default service timeout, which I think is something wacky like half an hour. As long as the user doesn't kill the kernel at that time, all should be well. Of course, if it were me, and it looked frozen for more than a couple minutes, I'd probably restart. )
+* Multiple in-flight retries are allowed.
 
 ### Desired Behavior
-* Prevent multiple in-flight retries to prevent the user from wasting their own resources (and the queues resources)
-* Add retry_count to retried jobs as well to aid in more book-keeping in a new field called `retry_number`
-* Non blocking job submission for submitting multiple jobs, possibly via using `run_job_batch` (requires refactor of run_job_batch)
-* One single submission to HTCondor instead of multiple job submissions
-* Ability to gracefully handle jobs with children
-* Ability to handle database consistentcy during retry failure
+* Create a retry_jobs field, and a toggle to have a list of retried jobs and to ensure DB consistency ( Ability to handle database consistency during retry failure. (see thread https://github.com/kbase/execution_engine2/pull/383#discussion_r638341940))
+* Add failure conditions in the `run()` method and see if any of those should be checked prior to starting jobs
+* Prevent retry if resolving `retry_parent` results in two or more of the same job_ids in a retry_jobs request (see thread https://github.com/kbase/execution_engine2/pull/383#discussion_r640907736)  ( Or silently just run the one retry job and report the results to both positions in the list )
 
+### Can be done with RunJob changes
+* * Non blocking job submission for submitting multiple jobs, possibly via using `run_job_batch` (requires refactor of run_job_batch)
+* * One single submission to HTCondor instead of multiple job submissions
 
-### Questions
+#### Won't do
+* Prevent multiple in-flight retries of the same original job to prevent the user from wasting their own resources (and the queues resources)
+* Add retry_number field
 
-#To Be Answered
+## New priority
+* Create a retry_jobs field, and expose list in api, and a T/F completeness toggle
```

Comment:
From what I recall the items were:
1. add a `completed_processing` (or whatever) boolean flag on the retry child job that starts out `False`
2. the retry child job should always have a retry parent field
3. the retry child job should have a batch parent field if it's part of a batch job
4. after creating the retry job, its ID will be added to a list of retry children on the retry parent using `$addToSet`
5. after creating the retry job, its ID will be added to a list of batch children on the batch parent using `$addToSet` if applicable
6. after that, set `completed_processing` to `True`

This means data integrity can be checked and corrected by scanning for jobs in an end state (complete / error / terminated) and doing steps 4-6 for each job.